velox/exec/fuzzer/AggregationFuzzerBase.cpp

```diff
@@ -353,18 +348,63 @@ std::vector<RowVectorPtr> AggregationFuzzerBase::generateInputDataWithRowNumber(
           generator->generate(signature.args, vectorFuzzer_, rng_, pool_.get());
     }
 
-    // Number of partitions is randomly generated and is at least 1.
-    auto numPartitions = size ? randInt(1, size) : 1;
-    auto indices = vectorFuzzer_.fuzzIndices(size, numPartitions);
-    auto nulls = vectorFuzzer_.fuzzNulls(size);
+    // Number of peer groups is randomly generated and is at least 1. Each input
+    // row is randomly mapped to a peer group using fuzzer generated indices.
+    // The total number of partitions is randomly generated such that it is
+    // between 1 and the number of peer groups. Each peer group is then mapped
+    // to a partition randomly using fuzzer generated indices, each partition
+    // would therefore contain rows belonging to one or more peer groups. Rows
+    // belonging to the same peer group will have the same value for the sorting
+    // keys, and rows belonging to the same partition will have the same value
+    // for the partitioning keys. For instance, the generated input data with 6
+    // rows could look as follows:
+    // numRows = 6, numPeerGroups = 3, numPartitions = 2,
+    // columns = {p0: VARCHAR, s0: INTEGER}, partitioningKeys = {p0},
+    // sortingKeys = {s0}
+    //   row1: 'APPLE'   2
+    //   row2: 'APPLE'   2
+    //   row3: 'APPLE'   2
+    //   row4: 'APPLE'   8
+    //   row5: 'ORANGE'  5
+    //   row6: 'ORANGE'  5
+    // In the above example, the sets of rows belonging to the same peer group
+    // are {row1, row2, row3}, {row4}, and {row5, row6}. The sets of rows
+    // belonging to the same partition are {row1, row2, row3, row4} and
+    // {row5, row6}. Generating input data with peer rows helps test window
+    // ranking functions effectively with the fuzzer. For instance, rank()
+    // requires rows in the same peer group have the same rank, this can be
+    // tested with the fuzzer by ensuring there are peer rows in the input data.
+    auto numPeerGroups = size ? randInt(1, size) : 1;
+    auto sortingIndices = vectorFuzzer_.fuzzIndices(size, numPeerGroups);
+    auto rawSortingIndices = sortingIndices->as<vector_size_t>();
+    auto sortingNulls = vectorFuzzer_.fuzzNulls(size);
+
+    auto numPartitions = randInt(1, numPeerGroups);
+    auto partitionIndices =
```

Comment:
Move the definitions of (raw)partitionIndices after (raw)peerGroupToPartitionIndices as that is closer to when they are used.